### ISSUE TITLE
Issue List

### ISSUE THREAD

**Issue by untr0py** [dzhu622911/nn-keras-cwrnn#1]
I've made a list of things to look at in cwrnn_keras.py:

- `class ClockworkRNNCell(layer):` There is nothing imported called "layer"

- `class ClockworkRNN(RNN):` There is nothing imported called "RNN"

- `valid_column = states[2][timestep]` timestep will be an array when there is more than 1 sample, so it can't be used as an index (I think)

- There appears to be no consistency with the size of the states. `self.state_size = self.units` indicates that it should be 1-dimensional. `return output, [output, timestep + 1]` indicates that it should be 2-dimensional, and `valid_column = states[2][timestep]` indicates that it should be 3-dimensional

- `output = K.concatenate([output, prev_output[:,valid_column:]], axis = 1)` This concatenation appears to increase the size of 'output' with each step.

- There are mixed tabs and spaces

I'm guessing that you haven't been able to run the code yet.

**Comment by dzhu622911** [comment]
@DeanIsMe, thanks a lot for your review,  and Yes, the code hasn't run yet, because there is another important work in my hand, and seems i can't come back to clockwork in few months, but I think i will try to run it in the weekend. :)

I review the code, and i think need todo below change in next verison:
1. `valid_column = states[2][timestep]` should be `valid_column =constants[0]`
2.  I really misunderstand of the self.state_size, i think it should be `self.state_size = [self.units, 1]`
3. for output, i will check when run, but i think when to compute:
```
        output = h + K.dot(prev_output, (self.recurrent_kernel * self.utm_mask)[:,:valid_column])
        if self.activation is not None:
            output = self.activation(output)
```
the shape of output is (batch, valid_column) , so after concatenate with prev_output:
`output = K.concatenate([output, prev_output[:,valid_column:]], axis = 1)`
the shape of output should be (batch, self.units) 

Thanks again for review my code. :+1:
